protocol/contracts/voting/MasterCandidate.sol

```diff
@@ -0,0 +1,191 @@
+
+        emit UpdateMetadata(msg.sender, discordId, discordHandle, twitterHandle);
+    }
+
+    function updateMetadata(
+        uint256 discordId,
+        string memory discordHandle,
+        string memory twitterHandle
+    ) external {
+        require(ownerOf(discordId) == msg.sender, "Only the candidate can update metadata");
+        
+        Candidate storage candidate = candidates[discordId];
+        candidate.discordHandle = discordHandle;
+        candidate.twitterHandle = twitterHandle;
+
+        emit UpdateMetadata(msg.sender, discordId, discordHandle, twitterHandle);
+    }
+
+    function toggleEndorsement(uint256 discordId) external {
+        _toggleEndorsement(msg.sender, discordId);
+    }
+
+    function toggleEndorsements(uint256[] memory discordIds) external {
+        _toggleEndorsements(msg.sender, discordIds);
+    }
+
+    /**
+     * @notice Withdraw for another user (gasless for the vault token holder)
+     * (assuming the owner has given authority for the caller to act on their behalf)
+     *
+     * @dev amount is explicit, to allow use case of partial vault withdrawals
+     */
+    function toggleEndorsementsFor(address account, uint256[] memory discordIds, uint256 deadline, uint8 v, bytes32 r, bytes32 s) public {
+        require(block.timestamp <= deadline, "MasterCandidate: expired deadline");
+
+        bytes32 structHash = keccak256(abi.encode(TOGGLE_ENDORSEMENTS_FOR_TYPEHASH, account, discordIds, deadline, _useNonce(account)));
+        bytes32 digest = _hashTypedDataV4(structHash);
+        address signer = ECDSA.recover(digest, v, r, s);
+
+        require(signer == account, "MasterCandidate: invalid signature");
+
+        _toggleEndorsements(account, discordIds);
+    }
+
+    function tokenURI(uint256 discordId) public view override returns (string memory) {
+        require(_exists(discordId), "ERC721Metadata: URI query for nonexistent token");
+
+        Candidate storage candidate = candidates[discordId];
+
+        // TODO(butlerji): Actually URI encode this somehow?
+        return string(abi.encodePacked(candidate.discordHandle, candidate.twitterHandle, discordId.toString()));
+    }
+
+    /**
+    * See {IERC20Permit-DOMAIN_SEPARATOR}.
+    */
+    // solhint-disable-next-line func-name-mixedcase
+    function DOMAIN_SEPARATOR() external view returns (bytes32) {
+        return _domainSeparatorV4();
+    }
+
+    function _toggleEndorsements(address account, uint256[] memory discordIds) internal {
+        for (uint256 i = 0; i < discordIds.length; i++) {
+            _toggleEndorsement(account, discordIds[i]);
+        }
+    }
+
+    function _toggleEndorsement(address account, uint256 discordId) internal {
+        require(_exists(discordId), "ElderCandidate: Cannot endorse nonexistent candidate");
+        endorsementsBy[account][discordId] = !endorsementsBy[account][discordId];
+
+        emit ToggleEndorsement(account, discordId, endorsementsBy[account][discordId]);
```

Comment:
Actually calculating the 'live vote' will be a combination of 3 APIs.

1/ token vault sub graph, which gives us the live token holding weighting.
2/ DAO Games API which gives us the live team weighting
3/ sub-graph which gives us the endorsements per account.

We can combine the 3 in the FE to show us a 'live tally' on who's currently a master.